projects/rocprofiler-systems/docs/how-to/openmp-profiling.rst

```diff
@@ -0,0 +1,198 @@
+
+Only a subset of OMPT callbacks are processed. The list of supported callbacks can be viewed by running the following command:
+
+.. code-block:: shell
+
+  rocprof-sys-run --help | awk '/--rocm-ompt-operations \[/,/\]/'
+
+.. note::
+
+  * ``omp_callback_functions`` is an internal callback used by ROCprofiler-SDK and should not be disabled.
+  * ``omp_parallel_begin`` and ``omp_parallel_end`` have been combined into a single track named ``omp_parallel``.
+  * ``omp_thread_end`` events are not supported.
+
+.. important::
+  If your system does not have a device to offload to, OMPT traces will not be generated.
+
+Configuration
+================
+
+To enable capturing of OMPT callbacks, the following parameter settings are required:
+
+.. code-block:: shell
+
+  ROCPROFSYS_USE_ROCM=ON
+  ROCPROFSYS_USE_OMPT=ON
+
+By default, ``ROCPROFSYS_USE_ROCM`` is enabled unless otherwise specified.
+
+.. tip::
+
+   Creating a default configuration file helps maintain consistent profiling settings across sessions.
+   For details, see the `configuring runtime options guide <https://github.com/ROCm/rocm-systems/blob/develop/projects/rocprofiler-systems/docs/how-to/configuring-runtime-options.rst>`_.
+
+Profiling a Fortran program that uses GPU offloading
+=====================================================
+
+.. tip::
+
+   The steps in this section also apply to C/C++ programs.
+
+For this example, we will be using the `Fortran Jacobi example <https://github.com/amd/HPCTrainingExamples/tree/main/Pragma_Examples/OpenMP/Fortran/8_jacobi/2_jacobi_targetdata>`_.
+This program uses OpenMP to offload compute to the GPU.
+
+Cloning and building the application
+---------------------------------------------
+
+Clone the repository and navigate to the folder containing the program:
+
+.. code-block:: shell
+
+  git clone https://github.com/amd/HPCTrainingExamples.git HPCTrainingExamples
+  cd HPCTrainingExamples/Pragma_Examples/OpenMP/Fortran/8_jacobi/2_jacobi_targetdata
+
+Compile the program:
+
+.. code-block:: shell
+
+  make FC=amdflang
+
+Collecting a trace with rocprof-sys-run
+------------------------------------------------
+
+Callback APIs, such as OMPT, can be traced using ``rocprof-sys-run``. To collect a trace for the Jacobi example, run the following command:
+
+.. code-block:: shell
+
+  ROCPROFSYS_USE_OMPT=ON rocprof-sys-run -- ./jacobi
+
+Once the command completes, a ``.proto`` file will be generated in the output directory:
+
+.. code-block:: shell
+
+  rocprofsys-jacobi-output/<timestamp>/
+
+By default, this file contains all captured traces from the profiling session.
+
+Understanding the proto file output
+-----------------------------------------------
+
+To view the generated ``.proto`` file in a browser, open the `Perfetto UI page <https://ui.perfetto.dev/>`_. Then, click on ``Open trace file`` and select the ``.proto`` file.
+The output should match:
```

Comment:
Traces may not exactly match because the system the user has access to may differ from yours. We can say instead, "The output trace would look like:"